### DISCUSSION TITLE
Gatsby Build Error.

### DISCUSSION THREAD

**Discussion by cy920820** [gatsbyjs/gatsby#36257] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

The compilation is normal under Mac OS system, and the compilation error is reported under Linux x64 system.

Specific error information:

![image](https://user-images.githubusercontent.com/21211512/181457829-cc5f1329-090f-4d42-b6f4-75a64b2e4c07.png)

### Reproduction Link

no

### Steps to Reproduce

1.
2.
3.
...

### Expected Result

Normal build

### Actual Result

Build error reporting on Linux x64 platform

### Environment

```shell
OS:  linux,x64
Node: v12.22.10
Npm: v6.14.16
Gatsby: v3.6.2
```

### Config Flags

const {languages, defaultLanguage} = require('./languages');

module.exports = {
  siteMetadata: {
    title: `barwaqt`,
    description: ``,
    author: ``
  },
  plugins: [
    `gatsby-plugin-react-helmet`,
    `gatsby-plugin-image`,
    {
      resolve: 'gatsby-plugin-web-font-loader',
      options: {
        google: {
          families: ['Poppins']
        }
      }
    },
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `gatsby-starter-default`,
        short_name: `starter`,
        start_url: `/`,
        background_color: `#663399`,
        theme_color: `#663399`,
        display: `minimal-ui`,
        icon: `src/images/64-64.png`, // This path is relative to the root of the site.
      },
    },
    `gatsby-plugin-gatsby-cloud`,
    // this (optional) plugin enables Progressive Web App + Offline functionality
    // To learn more, visit: https://gatsby.dev/offline
    `gatsby-plugin-offline`,
    `gatsby-plugin-stylus`,
    // {
    //   resolve: "gatsby-plugin-stylus",
    //   options: {
    //     use: [rupture()],
    //   },
    // },
    // {
    //   resolve: `gatsby-plugin-stylus`,
    //   options: {
    //       postCssPlugins: [somePostCssPlugin()],
    //   }
    // }
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/locales`,
        name: `locale`
      }
    },
    {
      resolve: `gatsby-plugin-react-i18next`,
      options: {
        localeJsonSourceName: 'locale',
        languages,
        defaultLanguage,
        // if you are using Helmet, you must include siteUrl, and make sure you add http:https
        siteUrl: `https://www.barwaqtfintec.com/`,
        // if you are using trailingSlash gatsby config include it here, as well (the default is 'always')
        // you can pass any i18next options
        i18nextOptions: {
          lowerCaseLng: true,
          saveMissing: false,
          interpolation: {
            escapeValue: false // not needed for react as it escapes by default
          },
          keySeparator: ".",
          nsSeparator: false
        },
        pages: [
          {
            matchPath: '/oneBill',
            languages: ['en']
          },
          {
            matchPath: '/RepaymentByEasypaisa',
            languages: ['en']
          },
          {
            matchPath: '/ApplicationProcess',
            languages: ['en']
          },
          {
            matchPath: '/repaymentViabyWa',
            languages: ['en']
          },
          {
            matchPath: '/repaymentViabyEA',
            languages: ['en']
          },
          {
            matchPath: '/404',
            languages: ['en']
          }
        ]
      }
    }
  ],
}

**Comment by cy920820** [comment, marked as answer]
“VipsOperation: class "webpsave_buffer" not found”

What's the problem?

**Reply by pieh** [reply]
Image transformation (via `gatsby-plugin-sharp`) is using `libvips` under the hood which is native binary. `sharp` should come with those binaries included, but it will use globally installed one if it exists (that can cause version mismatch if globally installed version is not matching what `sharp` want to use.

To test this hypothesis - can you (on Linux) do:
```
npm list sharp
```
that should show which version of `sharp` is used.

Next create some temporary directory, init empty npm project and install just that version of sharp:
```
mkdir sharp-test
cd sharp-test
npm init -y
npm install sharp@<version-from-npm-list>
```

And paste the output of the install command?

The line I'm somewhat expecting to see is something like:
```
info sharp Detected globally-installed libvips v8.9.2
```

If that's the case - options are:
 - update global libvips on system
 - use `SHARP_IGNORE_GLOBAL_LIBVIPS=1 npm install` on your site instead of regular installs
